packages/components/src/tools-panel/tools-panel/hook.js

```diff
@@ -101,6 +101,26 @@ export function useToolsPanel( props ) {
 		setMenuItems( items );
 	}, [ panelItems ] );
 
+	// Track whether optional controls, if any, are displayed or not.
+	const [
+		areOptionalControlsHidden,
+		setAreOptionalControlsHidden,
+	] = useState( false );
+
+	// Where no optional menu items are active, we display a plus icon
+	// to indicate the presence of further menu items.
+	useEffect( () => {
+		if ( menuItems.optional ) {
+			const optionalMenuItemsArray = Object.entries( menuItems.optional );
+			const newValue =
+				optionalMenuItemsArray.length > 0 &&
+				! Object.entries( menuItems.optional ).some(
+					( [ , isSelected ] ) => isSelected
+				);
+			setAreOptionalControlsHidden( newValue );
```

Comment:
Nit:

Since `Array.some(...)` works fine on an empty array fine, the `optionalMenuItemsArray.length > 0` check is not necessary.

This snippet could be simplified to

```suggestion
			setAreOptionalControlsHidden(
				! Object.entries( menuItems.optional ).some(
					( [ , isSelected ] ) => isSelected
				)
			);
```

Reply:
Thanks for the suggestion. 

We want to track whether optional items are available at all and if so, whether are they hidden.

We want to also return false for the case where there are no optional items. Where there are only "default" items, no controls are toggable* (is that a word?) 😄 

Unfortunately I named the var and set method a little too opaquely to make this clear.  It probably should be something like `setAreOptionalControlsAvailableAndHidden`

* _Toggly_ or _togglesworth_ seems like nice alternatives

Reply:
You got me thinking though, what if we show the plus icon as long as there are hidden options left to "unhide"_, e.g.,

![plus-icon-every](https://user-images.githubusercontent.com/6458278/133524261-ca9f1e16-f505-4b25-a83d-2e9c5d879506.gif)

Then we could use `every`:

```js
			setAreOptionalControlsAvailableAndHidden(
				! Object.entries( menuItems.optional ).every(
					( [ , isSelected ] ) => isSelected
				)
			);
```

Reply:
Thank you @ramonjd for detailing the logic — my bad, I had missed some parts of it!

> what if we show the plus icon as long as there are hidden options left to "unhide",

That is also a good alternative. Both approaches seem reasonable — _maybe_ the original approach seems to make more sense to me (but it may be just because I already got accustomed that logic?)

> but not as long as some Welsh train station names

Yeah, not sure if we should call our variables `Llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch`